### PR TITLE
Update Glossary based on recent decisions.

### PR DESCRIPTION
**Pull request by toddjonker**
### Description of changes:

See #216 for related changes.

----
_**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**_

### CODE REVIEW

File: src/glossary.adoc [hunk 1]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
-argument cardinality::
-The number of arguments to a macro invocation associated with a particular parameter.
-Since an argument produces a stream of values, this can be different from the parameter's _stream
-cardinality_.
-
 argument group::
-In a macro invocation, a sequence containing multiple arguments, used to express parameters that
-accept more than one argument.
+The concrete syntax for a _grouped parameter_.
+In a macro invocation, a list containing multiple arguments, delimited  explicitly with `[...]`
```
Comment (zslayton) [inline src/glossary.adoc]:
```suggestion
In a macro invocation, a list containing multiple arguments, delimited explicitly with `[...]`
```

File: src/glossary.adoc [hunk 2]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
+notation. Used to express parameters that accept more than one argument.
 
-cardinality constraint::
-The part of a macro signature that specifies a parameter's acceptable _argument cardinality_ and
-_stream cardinality_.
+cardinality::
+Describes the number of values that a parameter will accept when invoked.
```
Comment (zslayton) [inline src/glossary.adoc]: Dunno if this is common PL parlance, but
> ...that a parameter will accept when invoked.

sounds like the parameter itself is being invoked.

File: src/glossary.adoc [hunk 3]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
 A single sub-expression within a macro invocation, corresponding to one of the macro’s
-parameters.  This is by default a one-to-one relation, but a parameter's cardinality constraint
+parameters.  This is by default a one-to-one relation, but a parameter's grouping mode
 can change this to a many-to-one relation.
 
-argument cardinality::
-The number of arguments to a macro invocation associated with a particular parameter.
-Since an argument produces a stream of values, this can be different from the parameter's _stream
-cardinality_.
-
```
Comment (zslayton) [inline src/glossary.adoc:20]: 👍

File: src/glossary.adoc [hunk 4]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
+notation. Used to express parameters that accept more than one argument.
 
-cardinality constraint::
-The part of a macro signature that specifies a parameter's acceptable _argument cardinality_ and
-_stream cardinality_.
+cardinality::
+Describes the number of values that a parameter will accept when invoked.
+One of zero-or-one, exactly-one, zero-or-more, or one-or-more.
+Specified in a signature by one of the modifiers `*?*`, `*+*`, `*{asterisk}*`, `*+*`, `*\...*`
+or `*\...+*`.
```
Comment (zslayton) [inline src/glossary.adoc]: One of those `+`s should be a `!`.
```suggestion
One of zero-or-one, exactly-one, zero-or-more, or one-or-more.
Specified in a signature by one of the modifiers `*?*`, `*!*`, `*{asterisk}*`, `*+*`, `*\...*`
or `*\...+*`.
```

File: src/glossary.adoc [hunk 5]
```diff
@@ -54,8 +52,8 @@ The _symbol table directive_ is effectively a less capable alternative syntax.
 
 encoding environment::
 The context-specific data maintained by an Ion implementation while encoding or decoding data. In
-Ion 1.0 this consists of the local symbol table; in Ion 1.1 this is expanded to include the local module table
-and a collection of available modules.
+Ion 1.0 this consists of the current symbol table; in Ion 1.1 this is expanded to include the Ion
```
Comment (zslayton) [inline src/glossary.adoc]: Slightly more explicit:
```suggestion
Ion 1.0 this consists of the current symbol table; in Ion 1.1 this is expanded to also include the Ion
```

**Review by zslayton — APPROVED** [review]

File: src/glossary.adoc (hunk 2 above)
Comment (toddjonker) [reply src/glossary.adoc]: Eh, no, its lazy. Better to say "when the macro is invoked".

File: src/glossary.adoc [hunk 6]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
-accept more than one argument.
+The concrete syntax for a _grouped parameter_.
+In a macro invocation, a list containing multiple arguments, delimited explicitly with `[...]`
+notation. Used to express parameters that accept more than one argument.
 
-cardinality constraint::
-The part of a macro signature that specifies a parameter's acceptable _argument cardinality_ and
-_stream cardinality_.
+cardinality::
+Describes the number of values that a parameter will accept when invoked.
```
Comment (toddjonker) [inline src/glossary.adoc]:
```suggestion
Describes the number of values that a parameter will accept when the macro is invoked.
```

File: src/glossary.adoc [hunk 7]
```diff
@@ -10,21 +10,19 @@ A macro can be _fixed arity_ or _variable arity_.
...
-The number of arguments to a macro invocation associated with a particular parameter.
-Since an argument produces a stream of values, this can be different from the parameter's _stream
-cardinality_.
-
 argument group::
-In a macro invocation, a sequence containing multiple arguments, used to express parameters that
-accept more than one argument.
+The concrete syntax for a _grouped parameter_.
+In a macro invocation, a list containing multiple arguments, delimited explicitly with `[...]`
+notation. Used to express parameters that accept more than one argument.
```
Comment (almann) [inline src/glossary.adoc]:
```suggestion
notation in Ion text. Used to express parameters that accept more than one argument.
```

**Review by almann — APPROVED** [review]
🚢 LGTM, minor suggestion below.

**Pull request merged**